mlflow/types/chat.py

```diff
@@ -0,0 +1,130 @@
+from __future__ import annotations
+
+from typing import Literal, Optional, Union
+
+from pydantic import BaseModel as _BaseModel
+from pydantic import Field
+from typing_extensions import Annotated
+
+from mlflow.utils import IS_PYDANTIC_V2_OR_NEWER
+
+
+class BaseModel(_BaseModel):
+    @classmethod
+    def validate(cls, obj: any):
+        if IS_PYDANTIC_V2_OR_NEWER:
+            return cls.model_validate(obj)
+        else:
+            return cls.parse_obj(obj)
+
+
+class TextContentPart(BaseModel):
+    type: Literal["text"]
+    text: str
+
+
+class ImageUrl(BaseModel):
+    url: str  # either URL of an image, or bas64 encoded data
+    detail: Literal["auto", "low", "high"]
+
+
+class ImageContentPart(BaseModel):
+    type: Literal["image_url"]
+    image_url: ImageUrl
+
+
+class InputAudio(BaseModel):
+    data: str  # base64 encoded data
+    format: Literal["wav", "mp3"]
+
+
+class AudioContentPart(BaseModel):
+    type: Literal["input_audio"]
+    input_audio: InputAudio
+
+
+ContentPartsList = Annotated[
+    list[
+        Annotated[
+            Union[TextContentPart, ImageContentPart, AudioContentPart], Field(discriminator="type")
+        ]
+    ],
+    Field(min_items=1),
```

Comment:
Do we need `min_items=1`? The following query is useless but works fine.


```
$curl https://api.openai.com/v1/chat/completions \
  -H "Content-Type: application/json" \
  -H "Authorization: Bearer $OPENAI_API_KEY" \
  -d '{
    "model": "gpt-4o",
    "messages": [
      {
        "role": "user",
        "content": []
      }
    ],
    "max_tokens": 300
  }'
```

Reply:
hmm i guess we don't, i'll remove it. weird that it's an error for tools though:

```
curl https://api.openai.com/v1/chat/completions \
  -H "Content-Type: application/json" \
  -H "Authorization: Bearer $OPENAI_API_KEY" \
  -d '{
    "model": "gpt-4o",
    "messages": [
      {
        "role": "user",
        "content": []
      }
    ],
    "tools": [],
    "max_tokens": 300
  }'
{
  "error": {
    "message": "Invalid 'tools': empty array. Expected an array with minimum length 1, but got an empty array instead.",
    "type": "invalid_request_error",
    "param": "tools",
    "code": "empty_array"
  }
}
```